### PR TITLE
Allow acknowledgement_history#{author,is_sticky,is_persistent} being NULL

### LINKED ISSUE #305: Removing an acknowledgement crashes Icinga DB
When deleting an acknowledgement in the web interface, the latest Icinga DB master crashes:

```
2021-07-01T13:49:26.144Z	FATAL	icingadb/main.go:211	Error 1048: Column 'is_sticky' cannot be null
can't perform "INSERT INTO acknowledgement_history (environment_id,object_type,service_id,cleared_by,is_sticky,is_persistent,clear_time,comment,expire_time,host_id,set_time,endpoint_id,id,author) VALUES (:environment_id,:object_type,:service_id,:cleared_by,:is_sticky,:is_persistent,:clear_time,:comment,:expire_time,:host_id,:set_time,:endpoint_id,:id,:author) ON DUPLICATE KEY UPDATE clear_time = :clear_time,cleared_by = :cleared_by"
```

Setting an acknowledgement works fine, but when removing it, the above error happens. Maybe the event message has fewer JSON keys set? (But that's just speculation, haven't investigated further.)

### PR DESCRIPTION
**Pull request by Al2Klimov**
... for the case an ack_set event happened before Icinga DB history recording.

fixes #305
closes #318

### CODE REVIEW

**Comment by Al2Klimov** [comment]
## Purrs

```
*************************** 5. row ***************************
            id: �j
纽��0!������<�
2U��`���ent_id: �9��^kK
   endpoint_id: _�4p�Xfk%���)��%�
   object_type: service
       host_id: _�4p�Xfk%���)��%�
    service_id: �|?[{i-E��c0�����
      set_time: 1628076366135
    clear_time: 1628076429915
        author: NULL
    cleared_by: me
       comment: NULL
   expire_time: NULL
     is_sticky: NULL
 is_persistent: NULL
*************************** 6. row ***************************
            id: ��]��9�����
�u�p
2U��`���ent_id: �9��^kK
   endpoint_id: _�4p�Xfk%���)��%�
   object_type: service
       host_id: _�4p�Xfk%���)��%�
    service_id: ���@��A��m`�/�'�r�
      set_time: 1628076283776
    clear_time: 1628076296193
        author: me
    cleared_by: me
       comment: mine
   expire_time: NULL
     is_sticky: n
 is_persistent: n
```

**Comment by julianbrost** [comment]
I recommend using `mysql --binary-as-hex`

File: schema/mysql/schema.sql [hunk 1]
```diff
@@ -1013,12 +1013,12 @@ CREATE TABLE acknowledgement_history (
 
   set_time bigint unsigned NOT NULL,
   clear_time bigint unsigned DEFAULT NULL,
-  author varchar(255) NOT NULL COLLATE utf8mb4_unicode_ci,
+  author varchar(255) DEFAULT NULL COLLATE utf8mb4_unicode_ci COMMENT 'NULL if ack_set event happened before Icinga DB history recording',
   cleared_by varchar(255) DEFAULT NULL COLLATE utf8mb4_unicode_ci,
   comment text DEFAULT NULL,
```
Comment (lippserd) [inline schema/mysql/schema.sql:1018]: I would add the NULL info here too, as a comment is actually required.

**Review by lippserd — CHANGES_REQUESTED** [review]

File: schema/mysql/schema.sql (hunk 1 above)
Comment (Al2Klimov) [reply schema/mysql/schema.sql:1018]: Any idea why this column was already NULLable?

Comment (lippserd) [reply schema/mysql/schema.sql:1018]: Nope, no idea. Maybe you? Since you've introduced this table 😆

**Review by lippserd — APPROVED** [review]

**Pull request merged**